### PR TITLE
[ENH] splitter that replicates `loc` of another splitter

### PR DESCRIPTION
**Pull request by fkiraly**
This adds a splitter that takes a splitter and some data, and always replicates the same `loc` references for splits of another splitter.

Related issue and discussion: https://github.com/sktime/sktime/issues/4842

More general form of `temporal_train_test_split`, particularly useful for the case where we want to split `X` and `y`, and primarily `y`, and `X` can have different indices from `y`.

This ability partially addresses #4842, FYI @felipeangelimvieira

### CODE REVIEW

**Comment by fkiraly** [comment]
this is weird, equality tests fails only on windows, as array type seems to be `int64` vs `int`. Why only on windows?

File: sktime/forecasting/model_selection/_split.py [hunk 1]
```diff
@@ -1301,6 +1308,112 @@ def get_test_params(cls, parameter_set="default"):
...
+        for y_train_loc, y_test_loc in cv.split_loc(y_template):
+            y_train_iloc = y.get_indexer(y_train_loc)
+            y_test_iloc = y.get_indexer(y_test_loc)
+            yield y_train_iloc, y_test_iloc
+
+    def get_n_splits(self, y: Optional[ACCEPTED_Y_TYPES] = None) -> int:
+        """Return the number of splits.
+
+        Since this splitter returns a single train/test split,
+        this number is trivially 1.
```
Comment (benHeid) [inline sktime/forecasting/model_selection/_split.py:1369]: Same comment as in #4862 it's okay if it is answered there or only here:

I do not understand this comment, what is the referring to? The return value can be any positive integer depending on the parameters of the inner cv.

**Review by benHeid — CHANGES_REQUESTED** [review]
Only one comment, I am also fine if this comment is answered on #4862 since it is exactly the same comment

File: sktime/forecasting/model_selection/_split.py (hunk 1 above)
Comment (fkiraly) [reply sktime/forecasting/model_selection/_split.py:1369]: oh, that's a typo and wrong. Will fix.

**Review by benHeid — APPROVED** [review]
Thank you for addressing the comments :)

**Pull request merged**